types/lib/query-interface.d.ts

```diff
@@ -498,7 +498,7 @@ export class QueryInterface {
     tableName: TableName,
     records: object[],
     options?: QueryOptions,
-    attributes?: string[] | string
+    attributes?: object
```

Comment:
This isn't right  either actually.

I don't know if there's already a type for this or not.  At the top of my head, it should be moreso like:


```ts

type Attribute = Literal | string | [Literal | Fn | Col , string]

type Attributes = Array<Attribute> | { include?: Array<Attribute>, exclude: Array<Attribute> }

```

note: this is mostly psuedocode, I don't remember the exact typings for col, fn, and literals.

Reply:
@AllAwesome497 You are right, It isn't correct either. I think `ModelAttributes` from Model.d.ts seems to be appropriate here. What do you say?.
https://github.com/sequelize/sequelize/blob/23aa67e9abe440027aa07091beb35834e584e42e/types/lib/model.d.ts#L1358

Reply:
Such as this?

```ts
type Attribute<ModelAttr extends string> = Literal | Col | ModelAttr | [Literal | Fn | Col, ModelAttr];

type Attributes<> = Array<ModelAttr extends string = string> | { include?: Array<Attribute<ModelAttr>>, exclude: Array<Attribute<ModelAttr>> }
```

In usage:
```
attributes?: Attributes<keyof ModelAttributes>
```

If so, then I think that would be the ideal solution.